src/ucs/config/parser.c

```diff
@@ -817,10 +817,12 @@ ucs_status_t ucs_config_clone_array(const void *src, void *dest, const void *arg
     ucs_status_t status;
     unsigned i;
 
-    dest_array->data = ucs_calloc(src_array->count, array->elem_size,
-                                  "config array");
-    if (dest_array->data == NULL) {
-        return UCS_ERR_NO_MEMORY;
+    if (src_array->count) {
+        dest_array->data = ucs_calloc(src_array->count, array->elem_size,
```

Comment:
can dest_array->data be left un-initialized?

Reply:
If it's `src_array->count`, there is no point in initializing  `dest_array->data` - anyway `data` is not used, and its release is skipped (actually valgrind even complains on that, if doing `calloc()` wuth 0, then not `free()`ing it).

Reply:
1. maybe set it to NULL to avoid garbage?
2. `if (src_array->count > 0) {`

Reply:
Done